nussl/core/audio_signal.py

```diff
@@ -1653,73 +1654,99 @@ def embed_audio(self, ext='.mp3', display=True):
     #          Effect Hooks          #
     ##################################
 
-    def build_effects(self, reset=True, overwrite=False):
+    def apply_effects(self, reset=True, overwrite=False):
         """
-        Builds all effects in the AudioSignal's effects chains using the audio data of the calling 
-        AudioSignal. If reset=True, the current effects chains will be deleted. If overwrite=True, 
-        the calling AudioSignal's audio_data will be overwritten and the stft_data will be deleted. 
+        Applies all effects in the AudioSignal's effects chains using the audio data of the calling 
+        AudioSignal. If reset=True, the current effects chains will be deleted following applying 
+        them to the audio signal. If overwrite=True, the calling AudioSignal's audio_data will 
+        be overwritten and the stft_data will be deleted. 
 
         FFmpeg effects will be applied AFTER SoX effects. The effects time_stretch and pitch_shift 
         are SoX effects. All others are FFmpeg effects.
 
-        If no effects have been added to the AudioSignal's effects chains, then build_effects 
-        will throw a RuntimeError. 
+        If no effects have been added to the AudioSignal's effects chains, then apply_effects 
+        will return itself.
 
-        This will add a tremolo effect followed by a high pass effect to the AudioSignal's effects chains:
+        This will add a time stretch effect to the audio signal's effects chains. 
+
+        >>> signal.duration
+        10.0
+        >>> signal.time_stretch(0.5)
+        >>> signal.duration
+        10.0
+
+        However, the signal's duration hasn't changed! You will need to call apply_effects to make
+        the changes in the signal's effects chains. 
+
+        >>> new_signal = signal.apply_effects()
+        >>> new_signal.duration
+        5.0
+
+        You can also chain effects together. This will add a tremolo effect followed by a high pass
+        effect to the AudioSignal's effects chains:
 
         >>> audio_signal.tremolo(5, .6).high_pass(12000)
 
-        You can apply then apply the effects onto the AudioSignal with build_effects
+        Using overwrite here change the audio data of audio_signal, rather than create a new signal. 
+
+        >>> audio_signal.apply_effects(overwrite=True)
 
-        >>> audio_signal.build_effects(overwrite=True)
+        All effects applied will stored in `self.effects_applied`, a list of FilterFunctions
+        in the order that they were applied.
```

Comment:
I would also show the 1-liner version of this as icing on the cake: `new_signal = audio_signal.time_stretch(0.5).tremolo(5, .6).high_pass(12000).apply_effects()`